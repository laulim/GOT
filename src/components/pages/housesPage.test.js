import React from 'react';
import {shallow} from 'enzyme';
import HousesPage from './housesPage';

describe('Testing <HousesPage/>', () => {
  const housPage = shallow(<HousesPage/>);

  it('HousesPage has been rendered', () => {
    expect(housPage).toMatchSnapshot();
  });

  it('HousesPage state "error" is false', () => {
    expect(housPage.state().error).toBeFalsy();
  });

  it('HousesPage state "selectedHouse" is Null', () => {
    expect(housPage.state().selectedHouse).toBeNil();
  });

  it('Testing componentDidCatch, expect state "error" is true', () => {
    housPage.instance().componentDidCatch();
    expect(housPage.state().error).toBeTruthy();
  });

  it('Testing onItemSelected, expect state "selectedHouse" is equil to string "sdf8ds"', () => {
    const id = "sdf8ds";
    housPage.instance().onItemSelected(id);
    expect(housPage.state().selectedHouse).toBe("sdf8ds");
  })
})