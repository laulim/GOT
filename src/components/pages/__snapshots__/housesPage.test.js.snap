// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HousesPage/> HousesPage has been rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HousesPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <RandomItem
          btnName="house"
          fieldsList={
            Array [
              Array [
                "region",
                "Region",
              ],
              Array [
                "words",
                "Words",
              ],
              Array [
                "titles",
                "Titles",
              ],
              Array [
                "overlord",
                "Overlord",
              ],
              Array [
                "ancestralWeapons",
                "Ancestra wWeapons",
              ],
            ]
          }
          getData={[Function]}
          itemIdInterval={
            Array [
              80,
              1,
            ]
          }
        />,
        <RowBlock
          left={
            <ItemList
              getData={[Function]}
              onItemSelected={[Function]}
              pageDetails={
                Array [
                  45,
                  5,
                ]
              }
              renderItem={[Function]}
            />
          }
          right={
            <ItemDetails
              emptyTitle="house"
              getData={[Function]}
              itemId={null}
            >
              <Field
                field="region"
                label="Region"
              />
              <Field
                field="words"
                label="Words"
              />
              <Field
                field="titles"
                label="Titles"
              />
              <Field
                field="overlord"
                label="Overlord"
              />
              <Field
                field="ancestralWeapons"
                label="Ancestral weapons"
              />
            </ItemDetails>
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "btnName": "house",
          "fieldsList": Array [
            Array [
              "region",
              "Region",
            ],
            Array [
              "words",
              "Words",
            ],
            Array [
              "titles",
              "Titles",
            ],
            Array [
              "overlord",
              "Overlord",
            ],
            Array [
              "ancestralWeapons",
              "Ancestra wWeapons",
            ],
          ],
          "getData": [Function],
          "itemIdInterval": Array [
            80,
            1,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "left": <ItemList
            getData={[Function]}
            onItemSelected={[Function]}
            pageDetails={
              Array [
                45,
                5,
              ]
            }
            renderItem={[Function]}
          />,
          "right": <ItemDetails
            emptyTitle="house"
            getData={[Function]}
            itemId={null}
          >
            <Field
              field="region"
              label="Region"
            />
            <Field
              field="words"
              label="Words"
            />
            <Field
              field="titles"
              label="Titles"
            />
            <Field
              field="overlord"
              label="Overlord"
            />
            <Field
              field="ancestralWeapons"
              label="Ancestral weapons"
            />
          </ItemDetails>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <RandomItem
            btnName="house"
            fieldsList={
              Array [
                Array [
                  "region",
                  "Region",
                ],
                Array [
                  "words",
                  "Words",
                ],
                Array [
                  "titles",
                  "Titles",
                ],
                Array [
                  "overlord",
                  "Overlord",
                ],
                Array [
                  "ancestralWeapons",
                  "Ancestra wWeapons",
                ],
              ]
            }
            getData={[Function]}
            itemIdInterval={
              Array [
                80,
                1,
              ]
            }
          />,
          <RowBlock
            left={
              <ItemList
                getData={[Function]}
                onItemSelected={[Function]}
                pageDetails={
                  Array [
                    45,
                    5,
                  ]
                }
                renderItem={[Function]}
              />
            }
            right={
              <ItemDetails
                emptyTitle="house"
                getData={[Function]}
                itemId={null}
              >
                <Field
                  field="region"
                  label="Region"
                />
                <Field
                  field="words"
                  label="Words"
                />
                <Field
                  field="titles"
                  label="Titles"
                />
                <Field
                  field="overlord"
                  label="Overlord"
                />
                <Field
                  field="ancestralWeapons"
                  label="Ancestral weapons"
                />
              </ItemDetails>
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "btnName": "house",
            "fieldsList": Array [
              Array [
                "region",
                "Region",
              ],
              Array [
                "words",
                "Words",
              ],
              Array [
                "titles",
                "Titles",
              ],
              Array [
                "overlord",
                "Overlord",
              ],
              Array [
                "ancestralWeapons",
                "Ancestra wWeapons",
              ],
            ],
            "getData": [Function],
            "itemIdInterval": Array [
              80,
              1,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "left": <ItemList
              getData={[Function]}
              onItemSelected={[Function]}
              pageDetails={
                Array [
                  45,
                  5,
                ]
              }
              renderItem={[Function]}
            />,
            "right": <ItemDetails
              emptyTitle="house"
              getData={[Function]}
              itemId={null}
            >
              <Field
                field="region"
                label="Region"
              />
              <Field
                field="words"
                label="Words"
              />
              <Field
                field="titles"
                label="Titles"
              />
              <Field
                field="overlord"
                label="Overlord"
              />
              <Field
                field="ancestralWeapons"
                label="Ancestral weapons"
              />
            </ItemDetails>,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
